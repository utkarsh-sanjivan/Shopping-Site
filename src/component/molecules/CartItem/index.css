.product-container {
    padding: 10px;
    display: flex;
}

.product-name-container {
    width: 25%;
    display: flex;
    flex-direction: column;
}

.product-image {

}

.product-name {
    font-size: 12px;
    padding-top: 5px;
}

.cart-product-info {
    width: 25%;
    line-height: 70px;
    text-align: center;
    padding: 0 10px;
}

span {
    margin: 0 10px;
    cursor: pointer;
}