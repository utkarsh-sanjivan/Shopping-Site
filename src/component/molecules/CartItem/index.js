import React from 'react';
import './index.css';

export default function CartItem(props) {
    return (
        <div className='cart-product-container'>
            <div className='cart-product-name-container'>
                <img className='cart-product-image' src={props.image} width={50}/>
                <div className='cart-product-name'>{props.name}</div>
            </div>
            <div className='cart-product-info'>₹{props.price}</div>
            <div className='cart-product-info'>
                <span onClick={() => props.removeOne(props.id, props.cartQuantity, props.index)}>-</span>
                {props.cartQuantity}
                <span onClick={() => props.addOne(props.id)}>+</span>
            </div>
            <div className='cart-product-info'>₹{parseInt(props.price)*parseInt(props.cartQuantity)}</div>
        </div>
    );
}