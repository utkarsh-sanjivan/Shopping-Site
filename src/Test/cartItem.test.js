import React from 'react';
import { create } from "react-test-renderer";
import CartItem from '../component/molecules/CartItem/index';

let wrapper;
const renderComponent = props => <CartItem {...props} />;
const baseProps = {
    key: '1',
    image: '',
    name: 'Product1',
    description: 'lorem ipsum',
    price: '100.00',
    cartQuantity: '10',
    handleSelect: () => {},
};

beforeEach(() => {
  wrapper = create(renderComponent(baseProps));
});

describe('Testing Cart Item Component Rendering', () => {
  it('Cart Item Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
