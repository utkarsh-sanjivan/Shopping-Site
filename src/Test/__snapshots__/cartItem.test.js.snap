// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cart Item Component Rendering Cart Item Component should render properly 1`] = `
<div
  className="cart-product-container"
>
  <div
    className="cart-product-name-container"
  >
    <img
      alt="Cart Product"
      className="cart-product-image"
      src=""
      width={50}
    />
    <div
      className="cart-product-name"
    >
      Product1
    </div>
  </div>
  <div
    className="cart-product-info"
  >
    ₹
    100.00
  </div>
  <div
    className="cart-product-info"
  >
    <span
      onClick={[Function]}
    >
      -
    </span>
    10
    <span
      onClick={[Function]}
    >
      +
    </span>
  </div>
  <div
    className="cart-product-info"
  >
    ₹
    1000
  </div>
</div>
`;
